pkg/didcomm/crypto/legacy/authcrypt/decrypt.go

```diff
@@ -7,16 +7,165 @@ SPDX-License-Identifier: Apache-2.0
 package authcrypt
 
 import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
 	"errors"
+	"fmt"
 
+	"github.com/btcsuite/btcutil/base58"
+	chacha "golang.org/x/crypto/chacha20poly1305"
+	"golang.org/x/crypto/ed25519"
 	"golang.org/x/crypto/nacl/box"
 )
 
+// TODO implement legacy auth decrypt https://github.com/hyperledger/aries-framework-go/issues/294
+
 // Decrypt will decode the envelope using the legacy format
 // Using (X)Chacha20 encryption algorithm and Poly1035 authenticator
-func (c *Crypter) Decrypt(envelope []byte, recipientPrivKey *privateEd25519) ([]byte, error) {
-	// TODO implement legacy auth decrypt https://github.com/hyperledger/aries-framework-go/issues/294
-	return nil, errors.New("not implemented")
+func (c *Crypter) Decrypt(envelope []byte, recipientKey *keyPairEd25519) ([]byte, error) {
+	var envelopeData legacyEnvelope
+	err := json.Unmarshal(envelope, &envelopeData)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: see if there's a use for the n
+	protectedBytes, err := base64.URLEncoding.DecodeString(envelopeData.Protected)
+	if err != nil {
+		return nil, err
+	}
+
+	var protectedData protected
+	err = json.Unmarshal(protectedBytes, &protectedData)
+	if err != nil {
+		return nil, err
+	}
+	// TODO: do we keep this algorithm sanity check? Only commented out right now to accommodate a bug in ACA-Py
```

Comment:
Welcoming input here - we want to interop with ACA-Py, but we also want to meet the spec and thus interop with other hypothetical aries agents that decide not to support the incoming JWE format.

Reply:
What is ACA-PY using here?

(yes - we want to be able to interop with ACA-Py).

Reply:
the aim here to support backward compatibility with ACA-Py implementation.. the (new) jwe implementation takes care of properly handling the format required by the spec

Reply:
ACA-Py labels their messages as using xchacha20, while the algorithm used is always chacha20. They'll be updating to the spec once it's finalized, so I won't be keeping this check.